### ISSUE TITLE
The customer sign-in route CANNOT have relations

### ISSUE THREAD

**Issue by adevinwild** [medusajs/medusa#3557]
### Preliminary Checks

- [x] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/medusajs/medusa/issues

### Issue Summary

In the Store API section at the customer sign-in level, we can see that the client's delivery address can be retrieved in the case of authentication with the `expand` parameter.

But in the codebase, the deserialization of `expand` does not even exist.

I propose to either : 
- Remove the field from the documentation
- Add the functionality in the codebase

_Useful links_

> **Customer Auth**
> https://docs.medusajs.com/api/store#tag/Auth/operation/PostAuth

> Create Customer Session
> https://github.com/medusajs/medusa/blob/9997485c55cdf5e634aa24a9505f02cb5fe4c010/packages/medusa/src/api/routes/store/auth/create-session.ts

### How can this issue be resolved?

1. Either remove this field in the documentation
2. Or add the feature inside the `create-session.ts` file

### Are you interested in working on this issue?

- [X] I would like to fix this issue

**Comment by frankMagoba** [comment]
Hi @adevinwild , will the doc be updated or someone is already working on this functionality?

**Comment by adevinwild** [comment]
> Hi @adevinwild , will the doc be updated or someone is already working on this functionality?

Hi @frankMagoba 
I haven't heard anything yet, but as soon as I have an approval count on me to add it ^^

**Comment by dPreininger** [comment]
I would like to amend this issue with the related bug I found. In the Customer Store Controller, the expand query parameter is ignored in the code (FindOptions are always set as default options, at least in the GET /store/customers/me call). The docs claim, that you can expand the call to retrieve orders as well, however this is incorrect. I believe, that the intended functionality is to allow the use of expand query parameter but it was missed during implementation. Judging by the code, other query parameters, such as selecting fields should also be checked whether they work or not.

I am also willing to work on this if needed. @adevinwild maybe we could colab? You could fix the issue you found and I the one I did? We need to wait on what the core team thinks first tho.

Also: in my opinion, the auth endpoints would benefit if they had the added functionality of expandable relations (as is currently mentioned in the docs). I am not the one to make this decision tho.

**Comment by shahednasser** [comment]
Closing this issue as the new API reference adds a note about the availability of expanded relations. Any additional/separate issues can be added in a new issue.
